wallets/provider-xdefi/src/index.ts

```diff
@@ -100,22 +129,34 @@ export const canEagerConnect: CanEagerConnect = async ({ instance, meta }) => {
 };
 export const getWalletInfo: (allBlockChains: BlockchainMeta[]) => WalletInfo = (
   allBlockChains
-) => ({
-  name: 'XDefi',
-  img: 'https://raw.githubusercontent.com/rango-exchange/assets/main/wallets/xdefi/icon.svg',
-  installLink: {
-    CHROME:
-      'https://chrome.google.com/webstore/detail/xdefi-wallet/hmeobnfnfcmdkdcmlblgagmfpfboieaf',
-    BRAVE:
-      'https://chrome.google.com/webstore/detail/xdefi-wallet/hmeobnfnfcmdkdcmlblgagmfpfboieaf',
-    DEFAULT: 'https://xdefi.io/',
-  },
-  color: '#0646c7',
-  supportedChains: allBlockChains.filter((blockchainMeta) =>
-    [
-      ...SUPPORTED_ETH_CHAINS,
-      ...XDEFI_WALLET_SUPPORTED_NATIVE_CHAINS,
-      Networks.SOLANA,
-    ].includes(blockchainMeta.name as Networks)
-  ),
-});
+) => {
+  const supportedCosmosChains = cosmosBlockchains(allBlockChains)
+    .filter((blockchainMeta) => !!blockchainMeta.info)
+    .filter((blockchainMeta: BlockchainMeta) =>
+      SUPPORTED_COSMOS_CHAINS.includes(blockchainMeta.name as Networks)
```

Comment:
Could you combine these filters and eliminate unnecessary iteration?

Reply:
Thanks Fixed